### DISCUSSION TITLE
Why am I getting 404 warnings with `jupyter-widget` and `.ttf`/`.woff`/`.woff2` file extensions?

### DISCUSSION THREAD

**Discussion by paddyroddy** [quarto-dev/quarto-cli#10630] [Q&A]
### Description

quarto: `1.5.56`
macOS `14.6.1`

```sh
❯ quarto check
Quarto 1.5.56
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.56
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /Users/paddy/texlive/2024/bin/universal-darwin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.5
      Path: /private/tmp/talks/.venv/bin/python3
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.4.1
      Path: /opt/homebrew/Cellar/r/4.4.1/lib/R
      LibPaths:
        - /opt/homebrew/lib/R/4.4/site-library
        - /opt/homebrew/Cellar/r/4.4.1/lib/R/library
      knitr: 1.48
      rmarkdown: 2.28

[✓] Checking Knitr engine render......OK
```

Replicated https://quarto.org/docs/interactive/widgets/jupyter.html#leaflet-example

````qmd
---
author: Anonymous
date: 2024-08-28
format: revealjs
title: Demo
---

```{python}
from ipyleaflet import Map, Marker, basemaps, basemap_to_tiles
m = Map(
  basemap=basemap_to_tiles(
    basemaps.NASAGIBS.ModisTerraTrueColorCR, "2017-04-08"
  ),
  center=(52.204793, 360.121558),
  zoom=4
)
m.add_layer(Marker(location=(52.204793, 360.121558)))
m
```
````

I get these warnings using `quarto preview`, which I don't get on GitHub Actions (using `quarto publish`). Locally, if I run `quarto render` I don't think I get the warnings.
```sh
❯ quarto preview
Preparing to preview
[1/1] index.qmd

Starting python3 kernel...Done

Executing 'index.quarto_ipynb'
  Cell 1/1: ''...Done


Watching files for changes
Browse at http://localhost:5524/
GET: /
  /@jupyter-widgets/html-manager/dist/1551f4f60c37af51121f.woff2 (404: Not Found)
  /@jupyter-widgets/html-manager/dist/eeccf4f66002c6f2ba24.woff (404: Not Found)
  /@jupyter-widgets/html-manager/dist/be9ee23c0c6390141475.ttf (404: Not Found)
  /@jupyter-widgets/1551f4f60c37af51121f.woff2 (404: Not Found)
  /@jupyter-widgets/eeccf4f66002c6f2ba24.woff (404: Not Found)
  /@jupyter-widgets/be9ee23c0c6390141475.ttf (404: Not Found)
```

**Comment by mcanouil** [comment]
Could you properly format your message, especially the code? For instance, the code cell does not close.

Also, could you clarify exactly the context where you have the error?
You showed the result of `quarto preview) but you are talking about `quarto publish` and GitHub Actions.

**Reply by paddyroddy** [reply]
@mcanouil I don't know how to format as you suggest, what I have tried is 

![image](https://github.com/user-attachments/assets/aae5791f-19d8-41a9-b7e2-3d449a57a8f6)

If you know how to fix this, I'm all ears

**Reply by paddyroddy** [reply]
Looks like I made a typo, I get these warnings using `quarto preview`, not `quarto publish`/`quarto render`

**Reply by cscheid** [reply]
> If you know how to fix this, I'm all ears

You need more ticks in the outer code declaration, something like:

`````
Outer code:

````
some stuff.
inner code:

```
more stuff
```
````
`````

**Reply by paddyroddy** [reply]
Thanks, @cscheid! Updated

**Reply by mcanouil** [reply]
FYI, these guidelines were all there when you opened the discussion, see/read <https://github.com/quarto-dev/quarto-cli/discussions/new?category=q-a>.

**Reply by mcanouil** [reply]
I can reproduce the issue.
For now, I'm not exactly sure what is happening.

**Reply by mcanouil** [reply]
This issue only appears when using `quarto preview` and `revealjs`.
CC @cderv in case you have some insight.

**Reply by cderv** [reply]
> Locally, if I run quarto render I don't think I get the warnings.

I can see the warning too when I serve the rendered file. It seems we have some problem in how we handle Jupyter Widgets insertion inside revealjs presentation. I don't know why yet. 

Also it does not see we need to load those fonts when `format: html`, so maybe there is a specificity for revealjs we need to deal with.

**Reply by cderv** [reply]
@paddyroddy are these warning causing problem in how the leaflet is rendered ? 

Just to understand if this is an issue impacting output or just some content being loaded but may not be needed. 

It may also be an issue with Jupyter Widgets embeding with ipyleaflet, and not something with Quarto only.

**Reply by paddyroddy** [reply]
I don't think it is causing any issues, just disconcerting seeing warnings. As mentioned above, it doesn't seem to cause any issues with `publish`.

**Reply by cderv** [reply]
> As mentioned above, it doesn't seem to cause any issues with publish.

You won't see any log or warning from the command line. This is only happening on browser when file is loading. You see it with `preview` because a web server is opened and logging in terminal. 

Check your published document in a browser and open the network tab in Chrome Inspect mode. You may see the same problem.

**Reply by paddyroddy** [reply]
Yep looks like it's there too https://paddyroddy.github.io/talks/scipy-2024/#/2/1
![image](https://github.com/user-attachments/assets/81c1e983-8fa3-447a-8f36-25639a305594)

**Reply by cderv** [reply]
OK. So possibly a IPyleaflet issue when embedding. We would need to test it. Not sure yet why revealjs and html gets different ones though.

**Reply by lindyrock** [reply]
I'm getting this same issue.

````qmd
---
title: "Bug Report"
format:
  html:
    ipywidgets: embed  # Errors occur with or without this setting
execute:
  echo:
    false
---

```{python}
import ipywidgets as widgets

uploader = widgets.FileUpload()
uploader

```
````

<img width="783" alt="Screenshot 2025-02-09 at 8 06 24 PM" src="https://github.com/user-attachments/assets/265d093a-cc1b-4f70-ba17-457f6c2195c8" />

Quarto Version: 1.6.40
Operating System: MacOS 14.5

I noticed that the ipywidgets docs page is itself [unable to render widgets.Label](https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20List.html#label).
